app/Platform/Actions/UpdateGameAchievementsMetrics.php

```diff
@@ -20,16 +21,38 @@ public function execute(Game $game): void
         $playersHardcoreCalc = $playersHardcore ?: 1;
         $pointsWeightedTotal = 0;
         $achievements = $game->achievements()->published()->get();
+        if ($achievements->isEmpty()) {
+            return;
+        }
+
+        $achievementIds = $achievements->pluck('ID')->all();
+
+        // Get both total and hardcore counts in a single query.
+        $unlockStats = DB::table('player_achievements as pa')
+            ->leftJoin('UserAccounts as user', 'user.ID', '=', 'pa.user_id')
+            ->whereIn('pa.achievement_id', $achievementIds)
+            ->where('user.Untracked', false)
+            ->groupBy('pa.achievement_id')
+            ->select([
+                'pa.achievement_id',
+                DB::raw('COUNT(*) as total_unlocks'),
+                DB::raw('SUM(CASE WHEN unlocked_hardcore_at IS NOT NULL THEN 1 ELSE 0 END) as hardcore_unlocks'),
+            ])
+            ->get();
```

Comment:
Seems like this could be more eloquent-ized. I'm not sure if that would affect performance.
```
$unlockStats = PlayerAchievement::whereIn('achievement_id' $achievementIds)
     ->where('user', function ($query) { $query->where('Untracked', 0); })
     ->groupBy('achievement_id')
     ->select(...)
     ->get()
 ```

Reply:
On my local machine, this unfortunately did result in degraded performance. I believe the reason why Eloquent creates a subquery on the user stuff rather than using a left join.

However, I made some slight changes in latest to at least remove `DB::`.